include/aws/common/encoding.h

```diff
@@ -174,6 +192,15 @@ AWS_COMMON_API void aws_utf8_validator_reset(struct aws_utf8_validator *validato
  */
 AWS_COMMON_API int aws_utf8_validator_update(struct aws_utf8_validator *validator, struct aws_byte_cursor bytes);
 
+/**
+ * Update the validator with more bytes of text, using extra validation callback.
+ * Raises AWS_ERROR_INVALID_UTF8 if invalid UTF8 is encountered.
```

Comment:
Do we want to raise a separate error if validation fails (but it's valid utf-8)?